contracts/nois-gateway/src/state.rs

```diff
@@ -11,6 +11,8 @@ pub struct Config {
     pub manager: Addr,
     /// The price to pay in order to register the randomness job
     pub price: Coin,
+    /// The code ID of the payment contract to be instantatiated
+    pub payment_code_id: u64,
```

Comment:
> Any specific reason why this is not Option?

The idea is to prevent instantiating a gateway that would fail later on. In contrast to other cases, we don't need bidirectional dependencies. The payment contract can easily be deployed first.

> And u32 because we wouldn't expect to reach that many deployed codes. or does the cw factory instantiation use u64 thus no need to convert from u32 -> u64 upon payment factory instantiation?

u64 is the type that is used in the CosmWasm frame work for code IDs. If we use u32, we'd have to cast all the time when interacting with CosmWasm APIs.

Reply:
Right. payment does not beed to be instantiated by your script, only stored. So you can just move it up.

Reply:
That would create circurlar dependencies again. Let people create unused payment contracts if they want to. They don't hurt.